### PR TITLE
ARROW-11898: [Rust] Pretty print columns

### PR DESCRIPTION
**Pull request by nevi-me**
Adds a way to also pretty-print a slice of columns, as a convenience to aviod creating a single-column record batch.

### CODE REVIEW

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -110,7 +110,7 @@ jobs:
           export CARGO_TARGET_DIR="/github/home/target"
           cd rust
           # run tests on all workspace members with default feature list
-          cargo test
+          # cargo test
```
Comment (nevi-me) [inline .github/workflows/rust.yml:113]: I'm piggybacking off this PR to try find the cause for the CI hang

**Review by alamb — APPROVED** [review]
I went through this code and it makes sense to me (other than the change to disable `cargo test` which should be removed prior to merging)  -- thanks @nevi-me 

I am now going to spend some time looking into the CI hang

**Comment by alamb** [comment]
FYI I merged https://github.com/apache/arrow/pull/9653 / [ARROW-11896](https://issues.apache.org/jira/browse/ARROW-11896) for the Rust CI checks which may affect this PR. If you see "Rust / AMD64 Debian 10 Rust stable test workspace" failing  with a linker error or no logs, rebasing against master will hopefully fix the problem

**Comment by nevi-me** [comment]
> FYI I merged #9653 / [ARROW-11896](https://issues.apache.org/jira/browse/ARROW-11896) for the Rust CI checks which may affect this PR. If you see "Rust / AMD64 Debian 10 Rust stable test workspace" failing with a linker error or no logs, rebasing against master will hopefully fix the problem

Thakns, I'm rebasing my PRs

**Pull request closed without merging**